### ISSUE TITLE
TODO: Add title

### ISSUE THREAD

**Issue by ycdtosa** [koalaman/shellcheck#1534]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or 'online'): online
(but also tested with version 0.4.4 on debian)

- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
readonly Array=" --parameter=value --another=12 "
echo "${Array[@]}"
```

#### Here's what shellcheck currently says:
```

```
#### Here's what I wanted or expected to see:
```
Line 3	SCXXXX: variable used as an array but it was previosly assigned a string.
```
Note: The purpose of my real code is other than echo.

The description of the issue should say how to fix the code:
 ```sh
#!/bin/bash
readonly Array=( --parameter=value --another=12 )
echo "${Array[@]}"
```

I though this issue might be related, but it was fixed long ago #102 

This is kind of the opposite of SC2178

**Comment by Crestwave** [comment]
Note that a legitimate use of this is to make it expand to nothing if the variable is unset, as opposed to an empty string.
